File: shared/models/src/vault/Passkey.ts
/**
 * Passkey SQLite database type.
 */
export type Passkey = {
    /** The ID of the passkey */
    Id: string;

    /** The credential ID foreign key */
    CredentialId: string;

    /** The relying party identifier */
    RpId: string;

    /** The user ID provided by the relying party */
    UserId?: string | null;

    /** The public key */
    PublicKey: string;

    /** The private key */
    PrivateKey: string;

    /** The PRF encryption key associated with the passkey (optional, only set if PRF was requested by RP) */
    PrfKey?: Uint8Array | number[];

    /** The display name for the passkey */
    DisplayName: string;

    /** Additional data as JSON blob (Base64 encoded) */
    AdditionalData?: string | null;

    /** Created timestamp (epoch milliseconds) */
    CreatedAt: number;

    /** Updated timestamp (epoch milliseconds) */
    UpdatedAt: number;

    /** Soft delete flag (0/1) */
    IsDeleted: number;
}

File: apps/server/Databases/AliasClientDb/Passkey.cs
//-----------------------------------------------------------------------
// <copyright file="Passkey.cs" company="aliasvault">
// Copyright (c) aliasvault. All rights reserved.
// Licensed under the AGPLv3 license. See LICENSE.md file in the project root for full license information.
// </copyright>
//-----------------------------------------------------------------------

namespace AliasClientDb;

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using AliasClientDb.Abstracts;

/// <summary>
/// The Passkey class that stores WebAuthn/FIDO2 passkey information.
/// </summary>
public class Passkey : SyncableEntity
{
    /// <summary>
    /// Gets or sets the ID.
    /// </summary>
    [Key]
    public Guid Id { get; set; }

    /// <summary>
    /// Gets or sets the relying party identifier.
    /// </summary>
    [Required]
    [MaxLength(255)]
    public string RpId { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the user handle which represents the user ID provided by the relying party.
    /// </summary>
    [Required]
    public byte[] UserHandle { get; set; } = null!;

    /// <summary>
    /// Gets or sets the public key.
    /// </summary>
    public string PublicKey { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the private key.
    /// </summary>
    public string PrivateKey { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the PRF encryption key associated with the passkey (optional, only set if PRF was requested by RP).
    /// </summary>
    [MaxLength(64)]
    public byte[]? PrfKey { get; set; }

    /// <summary>
    /// Gets or sets the display name for the passkey.
    /// </summary>
    [MaxLength(255)]
    public string DisplayName { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the additional data as JSON blob.
    /// </summary>
    public byte[]? AdditionalData { get; set; }

    /// <summary>
    /// Gets or sets the item ID.
    /// </summary>
    public Guid ItemId { get; set; }

    /// <summary>
    /// Gets or sets the item object.
    /// </summary>
    [ForeignKey("ItemId")]
    public virtual Item Item { get; set; } = null!;
}
